src/base/bittorrent/session.cpp

```diff
@@ -3304,14 +3320,37 @@ bool Session::isListening() const
     return m_nativeSession->is_listening();
 }
 
-MaxRatioAction Session::maxRatioAction() const
+int Session::maxRatioActionIndex() const
 {
-    return static_cast<MaxRatioAction>(m_maxRatioAction.value());
+    switch (m_maxRatioAction) {
+    case Pause: return 0;
+    case Remove: return 1;
+    case DeleteFiles: return 2;
+    case DeleteFolder: return 3;
+    case EnableSuperSeeding: return 4;
+    default: return 0;
+    }
 }
 
-void Session::setMaxRatioAction(const MaxRatioAction act)
+void Session::setMaxRatioAction(const int comboIndex)
 {
-    m_maxRatioAction = static_cast<int>(act);
+    switch (comboIndex) {
+    case 0:
+    default:
+        m_maxRatioAction = Pause;
+        break;
+    case 1:
+        m_maxRatioAction = Remove;
+        break;
+    case 2:
+        m_maxRatioAction = DeleteFiles;
+        break;
+    case 3:
+        m_maxRatioAction = DeleteFolder;
+        break;
+    case 4:
+        m_maxRatioAction = EnableSuperSeeding;
```

Comment:
you should always put a `break;` in `case`.

Reply:
Coding style or something else?

Reply:
I'm saying you didn't place a `break;` below which you should add it.

Reply:
I got that, i'm asking whether it's for coding style or something else.

Reply:
> Coding style or something else?

It's good practice in case additional cases are added in the future. Reduces the likelihood of an error

Reply:
Ah ok thanks.